controller/src/beerocks/master/son_master_thread.cpp

```diff
@@ -671,24 +671,7 @@ bool master_thread::handle_cmdu_1905_channel_selection_response(Socket *sd,
 {
     auto mid = cmdu_rx.getMessageId();
     LOG(INFO) << "Received CHANNEL_SELECTION_RESPONSE_MESSAGE, mid=" << std::dec << int(mid);
-    int tlvType;
```

Comment:
Please add the same change to `son_slave_thread.cpp: handle_channel_selection_request`

Reply:
I already mention it on other comment but I guess you didn't see it.
This can be done if expecting to only one type of TLV. Wherever several TLV types can be received, the order of the TLVs is not defined in the standard.
Therefore, on the channel_selection_request handler, you can't use the do-while, because you expect to _receive channel_preference_ TLV and _transmit_power_limit_ TLV, in undefined order.

Reply:
OK I think I understand - please correct me if I'm wrong here -
on channel_selection_request handler, you expect specific TLVs, no unexpected ones - so you simply go over the buffer and add what you find out of the TLVs you expect. If you find a TLV which is not part of the message, such as a VS TLV that normally you would skip, it will be considered a failure, correct?

Reply:
I unresolved the conversation so that you will see my comment. If you approve my assumptions, then I agree with you of course.

Reply:
Kind of...
We can use do-while when we expect only **one** or more TLVsfrom the same type.
Such a thing applies to channel selection response and operating channel report.
Using do-while does not allow to skip unexpected TLVsuch as VS. If Unexpected TLV arrives, we will return failure.
On CMDUs that have **zero/one** or more TLV from different types, we must use a regular while loop, because we can't know which TLV will be next.
Such a thing applies to channel selection requests and channel preference report messages.
Regular while loop allows to handle unexpected TLVs such as VS or other and skip it.